R/visualize.R

```diff
@@ -0,0 +1,237 @@
+#' From the visualization, users could see the tradeoff between fairness metrics and accuracy.
+#' Those insights could help the users to choose the optimum model from their model sets.
+#' And the fairness consistency of the learners by visualizing the tradeoff for different resampling iterations.
+#' It could take multiple type of inputs, like ([PredictionClassif]), ([BenchmarkResult]) and ([ResampleResult]).
+#'
+#'
+#' @param object ([PredictionClassif])\cr ([BenchmarkResult])\cr ([ResampleResult])\cr
+#' The binary class prediction object that will be evaluated. Only one data task is allowed for BenchmarkResult or ResampleResult
+#' * If provided a ([PredictionClassif]). Then only one point will indicate the accuracy and fairness metrics for the current predictions.
+#' * If provided a ([ResampleResult]). Then the plot will compare the accuracy and fairness metrics for the same model, but different resampling iterations.
+#' * If provided a ([BenchmarkResult]). Then the plot will compare the accuracy and fairness metrics for all models and all resampling iterations.
+#'
+#' @param ...
+#' Arguments to be passed to methods. Such as:
+#'
+#' * `fairness_measure` ([Measure])\cr
+#' The fairness measures that will evaluated on object, could be single measure (`msr`) or list of measures (`msrs`), check [mlr3::Measure]. Default measure set to be \code{msr("fairness.fpr")}
+#'
+#' * `accuracy_measure` ([Measure])\cr
+#' The accuracy measure that will evaluated on object, could be single measure \link{msr} or measures \link{msrs}. Default measure set to be \code{\link[mlr3:MeasureClassif]{msr("classif.acc")}}
+#'
+#' * `task` ([TaskClassif])\cr
+#' The data task that contains the protected column, only required when the class of object is ([PredictionClassif])
+#'
+#'
+#'
+#' @examples
+#' library(mlr3fairness)
+#' library(mlr3learners)
+#' library(mlr3)
+#' library(ggplot2)
+#' library(data.table)
+#'
+#' # Setup the Fairness Measures and tasks
+#' task = tsk("adult_train")
+#' learner = lrn("classif.ranger", predict_type = "prob")
+#' learner$train(task)
+#' predictions = learner$predict(task)
+#' design = benchmark_grid(
+#'   tasks = tsk("adult_train"),
+#'   learners = lrns(c("classif.ranger", "classif.rpart"),
+#'                  predict_type = "prob", predict_sets = c("train", "test")),
+#'   resamplings = rsmps("cv", folds = 3)
+#' )
+#'
+#' bmr = benchmark(design)
+#' fairness_measure = msr("fairness.tpr")
+#' fairness_measures = msrs(c("fairness.tpr", "fairness.fnr"))
+#'
+#' fairness_accuracy_tradeoff(predictions, fairness_measure, task)
+#' fairness_accuracy_tradeoff(bmr, fairness_measure)
+#' @export
+fairness_accuracy_tradeoff <- function(object, ...){
+  UseMethod("fairness_accuracy_tradeoff")
+}
+
+#' @export
+fairness_accuracy_tradeoff.PredictionClassif <- function(object, fairness_measure = msr("fairness.fpr"), task, acc_measure = msr("classif.acc"), ...){
```

Comment:
Please add xlab and ylab with the name of the corresponding metric (not fairness but e.g. "fairness.fpr" etc)

Reply:
Done